src/article_metrics/ga_metrics/core.py

```diff
@@ -304,6 +297,56 @@ def query_ga_write_results(query, num_attempts=5):
     path = output_path_from_results(response)
     return response, write_results(response, path)
 
+
+# --- END GA3 LOGIC
+
+def output_path_v2(results_type, from_date_dt, to_date_dt):
+    """generates a path for results of the given type.
+    same logic as `output_path`, but more strict.
+    """
+    known_results_types = ['views', 'downloads',
+                           'blog-article', 'collection', 'digest', 'event', 'interview', 'labs-post', 'press-package']
+    ensure(results_type in known_results_types, "unknown results type %r: %s" % (results_type, ", ".join(known_results_types)))
+    ensure(isinstance(from_date_dt, datetime), "from_date_dt must be a datetime object")
+    ensure(isinstance(to_date_dt, datetime), "to_date_dt must be a datetime object")
```

Comment:
these work because a d cannot be an instance of a dt, but not the other way around. 

this is subtle nuance and I don't like it. switch to `type` checks